Get error when update vcpkg mujoco port to 3.2.6 and compile it.

### Intro

Hi!
I want to update mujoco to the latest release version on vcpkg (3.2.6)

### My setup

I am using wsl2 ubuntu 22.04. using gcc and mujoco 3.2.6

### What's happening? What did you expect?

I compile mujoco and get this error:

```
/home/tal/vcpkg/buildtrees/mujoco/src/3.2.6-7a23164287.clean/src/user/user_mesh.cc:1008:31: error: no match for ‘operator=’ (operand types are ‘std::vector<unsigned char>’ and ‘const std::vector<unsigned int>’)
 1008 |     num_face_vertices_ = mesh.num_face_vertices;
      |                               ^~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/vector:72,
                 from /usr/include/c++/11/functional:62,
                 from /usr/include/c++/11/pstl/glue_algorithm_defs.h:13,
                 from /usr/include/c++/11/algorithm:74,
                 from /home/tal/vcpkg/buildtrees/mujoco/src/3.2.6-7a23164287.clean/src/user/user_mesh.cc:15:
/usr/include/c++/11/bits/vector.tcc:198:5: note: candidate: ‘std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(const std::vector<_Tp, _Alloc>&) [with _Tp = unsigned char; _Alloc = std::allocator<unsigned char>]’
  198 |     vector<_Tp, _Alloc>::
      |     ^~~~~~~~~~~~~~~~~~~
/usr/include/c++/11/bits/vector.tcc:199:42: note:   no known conversion for argument 1 from ‘const std::vector<unsigned int>’ to ‘const std::vector<unsigned char>&’
  199 |     operator=(const vector<_Tp, _Alloc>& __x)
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
In file included from /usr/include/c++/11/vector:67,
                 from /usr/include/c++/11/functional:62,
                 from /usr/include/c++/11/pstl/glue_algorithm_defs.h:13,
                 from /usr/include/c++/11/algorithm:74,
                 from /home/tal/vcpkg/buildtrees/mujoco/src/3.2.6-7a23164287.clean/src/user/user_mesh.cc:15:
/usr/include/c++/11/bits/stl_vector.h:709:7: note: candidate: ‘std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(std::vector<_Tp, _Alloc>&&) [with _Tp = unsigned char; _Alloc = std::allocator<unsigned char>]’
  709 |       operator=(vector&& __x) noexcept(_Alloc_traits::_S_nothrow_move())
      |       ^~~~~~~~
/usr/include/c++/11/bits/stl_vector.h:709:26: note:   no known conversion for argument 1 from ‘const std::vector<unsigned int>’ to ‘std::vector<unsigned char>&&’
  709 |       operator=(vector&& __x) noexcept(_Alloc_traits::_S_nothrow_move())
      |                 ~~~~~~~~~^~~
/usr/include/c++/11/bits/stl_vector.h:730:7: note: candidate: ‘std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(std::initializer_list<_Tp>) [with _Tp = unsigned char; _Alloc = std::allocator<unsigned char>]’
  730 |       operator=(initializer_list<value_type> __l)
      |       ^~~~~~~~
/usr/include/c++/11/bits/stl_vector.h:730:46: note:   no known conversion for argument 1 from ‘const std::vector<unsigned int>’ to ‘std::initializer_list<unsigned char>’
  730 |       operator=(initializer_list<value_type> __l)
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
```

### Steps for reproduction

1. I updated vcpkg port of mujoco to 3.2.6
2. I added new dependencies port for mujoco (sdflib, marchingcubecpp)
3. I install mujoco via vcpkg: `./vcpkg install mujoco'

### Minimal model for reproduction

_No response_

### Code required for reproduction

_No response_

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

--- a/src/user/user_mesh.cc
+++ b/src/user/user_mesh.cc
@@ -61,7 +61,6 @@
 #include "user/user_objects.h"
 #include "user/user_resource.h"
 #include "user/user_util.h"
-#include <tiny_obj_loader.h>
 
 extern "C" {
 #include "qhull_ra.h"
@@ -394,7 +393,7 @@ void mjCMesh::CacheMesh(mjCCache* cache, const mjResource* resource,
       + (sizeof(int) * vertex_index_.size())
       + (sizeof(int) * normal_index_.size())
       + (sizeof(int) * texcoord_index_.size())
-      + (sizeof(unsigned char) * num_face_vertices_.size());
+      + (sizeof(face_vertices_type) * num_face_vertices_.size());
 
   std::shared_ptr<const void> cached_data(mesh, +[](const void* data) {
     const mjCMesh* mesh = static_cast<const mjCMesh*>(data);

--- a/src/user/user_objects.h
+++ b/src/user/user_objects.h
@@ -24,6 +24,7 @@
 #include <string_view>
 #include <utility>
 #include <vector>
+#include <tiny_obj_loader.h>
 
 #include <mujoco/mjtnum.h>
 #include <mujoco/mjmodel.h>
@@ -31,6 +32,8 @@
 #include <mujoco/mjspec.h>
 #include "user/user_cache.h"
 #include "user/user_util.h"
+
+using face_vertices_type = decltype(tinyobj::mesh_t::num_face_vertices)::value_type;
 
 // forward declarations of all mjC/X classes
 class mjCError;
@@ -993,7 +996,7 @@ class mjCMesh: public mjCMesh_, private mjsMesh {
   std::vector<int> vertex_index_;
   std::vector<int> normal_index_;
   std::vector<int> texcoord_index_;
-  std::vector<unsigned char> num_face_vertices_;
+  std::vector<face_vertices_type> num_face_vertices_;
 
   // compute the volume and center-of-mass of the mesh given the face center
   void ComputeVolume(double CoM[3], mjtGeomInertia gtype, const double facecen[3]);

